### DISCUSSION TITLE
New engine hat under development

### DISCUSSION THREAD

**Discussion by mairas** [hatlabs/discussions#69] [Show and tell]
The SH-ESP32 Engine Top Hat has been out of stock for a short while. Instead of ordering a new batch, I decided to design a new board which merges the functionality into the SH-ESP32.

The board will have same features as the original engine hat, but the connectors are better and there is proper isolation between the connectors and the main board. These changes will simplify using the board in common use cases. Optocoupler I/O has been removed.

The design is well under way. Apart from small tweaks, I just need to add silk screen labels before I can order the first prototypes.

Here's a small preview of the current design:

<img width="1470" alt="Screenshot 2023-09-02 at 20 51 45" src="https://github.com/hatlabs/discussions/assets/407010/9215f467-3251-4125-82ba-222ab2478fb4">

**Comment by tomekkosinski** [comment]
Hi,
I will love the new design. This is how my Engine Hat case looks after some "dremeling:
![IMG_5820](https://github.com/hatlabs/discussions/assets/4331414/b7771d13-555c-4b3b-ae34-ac3eb166f6df)


Having a better connector is a vast improvement. What is the reason for removing optocouplers?

**Reply by mairas** [reply]
Now that is a neat design! Kudos!

The optocouplers were removed partially due to space considerations and partially because HALMET (Hat Labs Marine Engine/Tank interface board, the name for the new design) is more focused on the analog and digital inputs. The SH-ESP32 will remain available for those who want to use it for other purposes.

HALMET does have a slightly different (and improved) isolation approach from the Engine Hat. The isolation barrier resides between the 4 digital + 4 analog inputs and the MCU. As long as other connections to external systems are not being made, it is safe to power the board using N2K.

**Comment by big-col** [comment]
Can't wait to get my hands on the new version!!

**Comment by wjn13** [comment]
First of all glad to find some nice products for professional looking interfaces. I am developing a NMEA2000 to analog converter (to display NMEA data on analog gauges), with the ttlappalainen NMEA2000 library. This works fine and I wanted to develop a board for this, when I found your ESP32 isolated CAN board SH-ESP32, this would suit my need for this part.
(I am refitting the boat with new electronics, all is NMEA2000 nowadays. So I decided to make NMEA2000 the main communication bus, and want to get rid of old thick cables.)

Now for the second part, I need to convert my old analog engine sensors to NMEA2000. Then you just developed HALMET, which should be good for this tasks.

Only question I have, in this version of HALMET there are only 8 inputs usable (isolated that is). 

If for example I want to monitor engine exhaust temperature, done by a thermocouple, most common way I found is to use a  MAX6675, I lose the isolation.
Same for the I2C bus, no isolation.

Question, do you plan to create a version of HALMET with isolated CAN bus connection, more like the SH-ESP32?
This will also reduce the LEN number on the NMEA2000 bus, if power to the board is supplied separetely.

Or alternatively produce a new batch of the Engine Top Hat, for those who want to keep the NMEA2000 bus isolated.

**Reply by mairas** [reply]
One of the design criteria for HALMET was to simplify the isolation issues for regular use cases. The engine hat couldn't really be powered through the N2K bus at all, and having the main inputs isolated simplifies the setup for most people.

I think you could still use a MAX6675 and a thermocouple and maintain isolation, as long as the thermocouple remains isolated from the engine. The same goes for I2C. As long as any I2C devices are only connected and powered by HALMET, they don't compromise N2K isolation.

**Reply by wjn13** [reply]
So probably in my case the SH-ESP32 makes more sense. I found some options to connect thermocouple with an analog output, but will have too less inputs on the HALMET for per engine. Other option is to isolate the I2C bus, for which modules are also available.
Hence my question if the Engine Top Hat will be produced (maybe in limited numbers), in this way I can just use SH-ESP32 as base for multiple purposes.
